server/src/test/java/org/candlepin/resource/HypervisorCloudProfileTest.java

```diff
@@ -0,0 +1,194 @@
+
+        result = hypervisorResource.hypervisorUpdate(
+            hostGuestMap, principal, owner.getKey(), true);
+
+        List<HypervisorConsumerDTO> updated = new ArrayList<>(result.getUpdated());
+
+        assertEquals(1, updated.size());
+        assertNotNull(consumerCurator.findByUuid(updated.get(0).getUuid()).getRHCloudProfileModified());
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnConsumerTypeUpdates() {
+        ConsumerType system = new ConsumerType(ConsumerType.ConsumerTypeEnum.SYSTEM);
+        consumerTypeCurator.create(system);
+
+        String hypervisorid = "test-host";
+        Consumer testConsumer = new Consumer(hypervisorid, someuser.getUsername(), owner, system);
+        HypervisorId hypervisorId = new HypervisorId(hypervisorid);
+        hypervisorId.setOwner(owner);
+        testConsumer.setHypervisorId(hypervisorId);
+        testConsumer = consumerCurator.create(testConsumer);
+
+        Map<String, List<GuestIdDTO>> hostGuestMap = new HashMap<>();
+        hostGuestMap.put(testConsumer.getName(), new ArrayList<>());
+
+        HypervisorUpdateResultDTO result = hypervisorResource.hypervisorUpdate(
+            hostGuestMap, principal, owner.getKey(), true);
+
+        List<HypervisorConsumerDTO> updated = new ArrayList<>(result.getUpdated());
+
+        assertEquals(1, updated.size());
+        assertNotNull(consumerCurator.findByUuid(updated.get(0).getUuid()).getRHCloudProfileModified());
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnGuestUpdate() {
+        Consumer testConsumer = new Consumer("test_consumer", someuser.getUsername(), owner, hypervisorType);
+        testConsumer = consumerCurator.create(testConsumer);
+
+        GuestIdDTO guest = TestUtil.createGuestIdDTO("some_guest");
+        GuestId guestEnt = new GuestId();
+        guestEnt.setId("some_id");
+
+        Date beforeUpdateTimestamp = consumerCurator.findByUuid(testConsumer.getUuid())
+            .getRHCloudProfileModified();
+
+        guestIdResource.updateGuest(testConsumer.getUuid(), guest.getGuestId(), guest);
+        Date afterUpdateTimestamp = consumerCurator.findByUuid(testConsumer.getUuid())
+            .getRHCloudProfileModified();
+
+        assertNotNull(afterUpdateTimestamp);
+        assertNotEquals(beforeUpdateTimestamp, afterUpdateTimestamp);
+    }
+
+    @Test
+    public void testCloudProfileUpdatedOnMultipleGuestUpdate() {
+        List<GuestIdDTO> guestIds = new LinkedList<>();
+        guestIds.add(TestUtil.createGuestIdDTO("1"));
+
+        Date beforeUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        guestIdResource.updateGuests(consumer.getUuid(), guestIds);
+        Date afterUpdateTimestamp = consumerCurator.findByUuid(consumer.getUuid())
+            .getRHCloudProfileModified();
+
+        assertNotNull(afterUpdateTimestamp);
+        assertNotEquals(beforeUpdateTimestamp, afterUpdateTimestamp);
+    }
+
```

Comment:
Same for this class, a couple tests for each of HypervisorResource and GuestIdResource where we verify the profile timestamp is not updated because it is a field we don't care about

Reply:
Added.